src/Install/Console/InstallCommand.php

```diff
@@ -121,232 +100,82 @@ protected function fire()
-
-        $this->info('Writing config');
 
-        file_put_contents(
-            $this->getConfigFile(),
-            '<?php return '.var_export($config, true).';'
-        );
-    }
-
-    protected function runMigrations()
-    {
-        $this->migrator->setOutput($this->output);
-        $this->migrator->getRepository()->createRepository();
-        $this->migrator->run(__DIR__.'/../../../migrations');
-    }
-
-    protected function writeSettings()
-    {
-        $settings = new DatabaseSettingsRepository($this->db);
-
-        $this->info('Writing default settings');
+        if ($validation->fails()) {
+            throw new Exception(implode("\n",
+                call_user_func_array('array_merge',
+                    $validation->getMessageBag()->toArray())));
+        }
 
-        $settings->set('version', $this->application->version());
+        $admin = $this->dataSource->getAdminUser();
 
-        foreach ($this->settings as $k => $v) {
-            $settings->set($k, $v);
+        if (strlen($admin['password']) < 8) {
+            throw new Exception('Password must be at least 8 characters.');
         }
-    }
-
-    protected function createAdminUser()
-    {
-        $admin = $this->adminUser;
 
         if ($admin['password'] !== $admin['password_confirmation']) {
             throw new Exception('The password did not match its confirmation.');
         }
 
-        $this->info('Creating admin user '.$admin['username']);
-
-        $uid = $this->db->table('users')->insertGetId([
-            'username' => $admin['username'],
-            'email' => $admin['email'],
-            'password' => (new BcryptHasher)->make($admin['password']),
-            'joined_at' => time(),
-            'is_email_confirmed' => 1,
-        ]);
-
-        $this->db->table('group_user')->insert([
-            'user_id' => $uid,
-            'group_id' => Group::ADMINISTRATOR_ID,
-        ]);
-    }
-
-    protected function enableBundledExtensions()
-    {
-        $extensions = new ExtensionManager(
-            new DatabaseSettingsRepository($this->db),
-            $this->application,
-            $this->migrator,
-            $this->application->make(Dispatcher::class),
-            $this->application->make('files')
-        );
-
-        $disabled = [
-            'flarum-akismet',
-            'flarum-auth-facebook',
-            'flarum-auth-github',
-            'flarum-auth-twitter',
-            'flarum-pusher',
-        ];
-
-        foreach ($extensions->getExtensions() as $name => $extension) {
-            if (in_array($name, $disabled)) {
-                continue;
-            }
-
-            $this->info('Enabling extension: '.$name);
+        if (! filter_var($admin['email'], FILTER_VALIDATE_EMAIL)) {
+            throw new Exception('You must enter a valid email.');
+        }
 
-            $extensions->enable($name);
+        if (! $admin['username'] || preg_match('/[^a-z0-9_-]/i',
```

Comment:
As we're using the Validation Factory, wouldn't it be better to have a Custom validation rule for usernames that adheres to our (regular expression) requirements?